Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt

```diff
@@ -0,0 +1,191 @@
+            val totalFiles = directory.walkFiles()
+            Assert.assertEquals(4, batchReadyFiles.size)
+            Assert.assertEquals(totalFiles.iterator().asSequence().toList().size, 5)
+            for (file in batchReadyFiles) {
+                Assert.assertEquals(eventStorage.eventsFor(file).size, batchSize)
+            }
+
+            val currentFile = directory
+                .walkFiles()
+                .filter { !it.name.endsWith("ready") }
+                .first()
+            Assert.assertEquals(eventStorage.eventsFor(currentFile).size, 2)
+        }
+    }
+
+    @Test
+    fun emitting_an_event_batches_all_batches_sent_cleaned_up() = runTest {
+        val eventStorage = EventStorageImpl(mockContext)
+        val testDispatcher = UnconfinedTestDispatcher(testScheduler)
+        val batchSize = 10
+        val flushPolicy = object : FlushPolicy {
+            private var size = 0
+            override fun reset() {
+                size = 0
+            }
+
+            override fun hit(event: Event) {
+                size++
+            }
+
+            override fun shouldFlush(): Boolean {
+                return size >= batchSize
+            }
+        }
+        var uploadRequestCount = 0
+        val uploader = object : EventSenderUploader {
+            override suspend fun upload(events: EventBatch): Boolean {
+                uploadRequestCount++
+                return true
+            }
+        }
+        val engine = EventSenderEngineImpl(
+            eventStorage,
+            clientSecret,
+            flushPolicies = listOf(flushPolicy),
+            dispatcher = testDispatcher,
+            uploader = uploader
+        )
+        eventSender = Confidence(
+            eventSenderEngine = engine,
+            dispatcher = testDispatcher,
+            diskStorage = mock(),
+            clientSecret = "",
+            flagApplierClient = mock(),
+            flagResolver = mock()
+        )
+        val eventSender = this@EventSenderIntegrationTest.eventSender
+        val eventCount = 4 * batchSize + 2
+        requireNotNull(eventSender)
+        repeat(eventCount) {
+            eventSender.send("eventDefinitions/navigate")
+        }
+        advanceUntilIdle()
+        Assert.assertEquals(uploadRequestCount, eventCount / batchSize)
+        runBlocking {
+            val batchReadyFiles = eventStorage.batchReadyFiles()
+            val totalFiles = directory.walkFiles()
+            // all files are sent and cleaned up
+            Assert.assertEquals(batchReadyFiles.size, 0)
+            // only current file exists
+            Assert.assertEquals(totalFiles.iterator().asSequence().toList().size, 1)
+
+            val currentFile = directory
+                .walkFiles()
+                .filter { !it.name.endsWith("ready") }
+                .first()
+            Assert.assertEquals(eventStorage.eventsFor(currentFile).size, 2)
+        }
+    }
+}
```

Comment:
I'm missing tests that validates that parent-child context inheritance + removal etc works. Or did I miss them?

Reply:
done.